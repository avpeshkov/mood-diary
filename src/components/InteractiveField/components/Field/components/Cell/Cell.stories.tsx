import React from "react";
import { action } from "@storybook/addon-actions";
import { withKnobs, text, number } from "@storybook/addon-knobs";
import { Cell } from "./Cell";
export default {
    title: "Cell",
    decorators: [withKnobs],
};

export const nonFilled = () => [
    <Cell onClick={action("Cell clicked")} x={number("x", 1)} y={number("y", 23)} key="jsx" />,
    <Cell onClick={action("Cell clicked")} x={number("x", 1)} y={number("y", 23)} key="jsx2" />,
];

export const filledWithX = () => [
    <Cell onClick={action("Cell clicked")} x={number("x", 1)} y={number("y", 23)} key="jsx">
        {text("filled with", "x")}
    </Cell>,
    <Cell onClick={action("Cell clicked")} x={number("x", 1)} y={number("y", 23)} key="jsx2">
        {text("filled with", "x")}
    </Cell>,
];

export const filledWithY = () => [
    <Cell onClick={action("Cell clicked")} x={number("x", 1)} y={number("y", 23)} key="jsx">
        {text("filled with", "y")}
    </Cell>,
    <Cell onClick={action("Cell clicked")} x={number("x", 1)} y={number("y", 23)} key="jsx2">
        {text("filled with", "y")}
    </Cell>,
];
