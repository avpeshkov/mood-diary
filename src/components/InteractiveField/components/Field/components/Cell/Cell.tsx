import React, { FC } from "react";
import type { CellProps } from "types/field";

import { CellItem } from "./CellItems";

export const Cell: FC<CellProps> = ({ children, x, y, onClick }) => {
    const isFilled = Boolean(children);

    return (
        <CellItem isFilled={isFilled} onClick={() => !isFilled && onClick(x || 0, y || 0)}>
            {children}
        </CellItem>
    );
};
