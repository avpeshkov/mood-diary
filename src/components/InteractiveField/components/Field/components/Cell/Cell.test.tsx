import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

import { Cell } from "./Cell";

describe("Cell", () => {
    it("renders button for empty cell", () => {
        expect(renderer.create(<Cell onClick={jest.fn()} />).toJSON()).toMatchSnapshot();
    });
    it("renders button for filled with x cell", () => {
        expect(renderer.create(<Cell onClick={jest.fn()}>x</Cell>).toJSON()).toMatchSnapshot();
    });
    it("calls onClick callback on click by empty cell", () => {
        const onClick = jest.fn();
        const wrapper = mount(<Cell onClick={onClick} />);
        wrapper.simulate("click");
        expect(onClick).toHaveBeenCalled();
    });
    it("does not call onClick callback on click by filled cell", () => {
        const onClick = jest.fn();
        const wrapper = mount(<Cell onClick={onClick}>y</Cell>);
        wrapper.simulate("click");
        expect(onClick).not.toHaveBeenCalled();
    });
    it("calls onClick callback with passed x, y params", () => {
        const onClick = jest.fn();
        const x = 12;
        const y = 14;
        const wrapper = mount(<Cell onClick={onClick} x={x} y={y} />);
        wrapper.simulate("click");
        expect(onClick).toHaveBeenCalledWith(x, y);
    });
});
