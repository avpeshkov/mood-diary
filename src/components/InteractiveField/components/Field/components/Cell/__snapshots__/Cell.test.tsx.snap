// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders button for empty cell 1`] = `
.emotion-0 {
  width: 25px;
  height: 25px;
  border: 1px solid;
  display: inline-block;
  border-radius: 10px;
  line-height: 25px;
  text-align: center;
  margin: 5px;
  vertical-align: bottom;
  border-color: gray;
}

<button
  className="emotion-0"
  onClick={[Function]}
/>
`;

exports[`Cell renders button for filled with x cell 1`] = `
.emotion-0 {
  width: 25px;
  height: 25px;
  border: 1px solid;
  display: inline-block;
  border-radius: 10px;
  line-height: 25px;
  text-align: center;
  margin: 5px;
  vertical-align: bottom;
  border-color: lightgray;
  color: gray;
}

<button
  className="emotion-0"
  onClick={[Function]}
>
  x
</button>
`;
